### PR TITLE
build: node 18 and update outdated

### PR DESCRIPTION
**Pull request by eldu**
- node 16 --> 18
- update outdated
- update pre commit

### CODE REVIEW

**Review by s-bessey — COMMENTED** [review]
It looks like the readme needs updating in a couple places

1. Now serves to 5173?
2. line 12 still indicates node 16+
3. `npm run tscheck` is listed on line 28 but I get an error attempting that command

File: README.md [hunk 1]
```diff
@@ -18,14 +18,13 @@ $ npm install
 
 ### Compiles and hot-reloads for development
 ```bash
-# serve with hot reload at localhost:3000
+# serve with hot reload at localhost:5173
```
Comment (anna-murphy) [inline README.md:21]: is there a reason this changed? On Provident I think this happened to me because I was using NPM when i should have been using Yarn.

**Review by anna-murphy — COMMENTED** [review]
Looks good!

File: README.md (hunk 1 above)
Comment (eldu) [reply README.md:21]: it's the default for vite. it use to be 3000
https://vitejs.dev/config/server-options.html

**Pull request merged**